### PR TITLE
[WIP] occt: update to 7.6.2

### PR DESCRIPTION
**Pull request by karl-nilsson**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x64-glibc

Like previously, broken ABI
```console
During initialization the error "/usr/lib/freecad/lib64/Part.so: undefined symbol: _ZN24BRepBuilderAPI_MakeShape5BuildEv" occurred in /home/nilsson/.FreeCAD/Mod/A2plus/InitGui.py                                                                                                                        
Please look into the log file for further information                                                                                                
Wizard shaft module cannot be loaded                                                                                                                 
/usr/lib/freecad/lib64/Part.so: undefined symbol: _ZN24BRepBuilderAPI_MakeShape5BuildEv                                                              
connect failed: No such file or directory      
```

Looks like we'll have to update or patch freecad
```console
In file included from /usr/include/opencascade/Extrema_POnSurf.lxx:15,
                 from /usr/include/opencascade/Extrema_POnSurf.hxx:75,
                 from /usr/include/opencascade/Extrema_ExtPElS.hxx:27,
                 from /usr/include/opencascade/Extrema_ExtPS.hxx:24,
                 from /usr/include/opencascade/ShapeAnalysis_Surface.hxx:20,
                 from ../src/3rdParty/salomesmesh/inc/SMESH_MesherHelper.hxx:37,
                 from ../src/3rdParty/salomesmesh/src/SMESH/SMESH_Mesh.cpp:34:
/usr/include/opencascade/Precision.hxx:98:7: note: previous definition of 'class Precision'
   98 | class Precision
      |       ^~~~~~~~~

```

ping @Piraty

### CODE REVIEW

**Comment by karl-nilsson** [comment]
I'm just going to wait for the release of freecad v0.20. There's a slew of patches required to build with occt 7.6.x

**Comment by luzpaz** [comment]
Thanks for you work on this!

**Comment by karl-nilsson** [comment]
@Piraty I think it's ready

@yopito I updated FreeCAD to v0.20, and it looks like most of the patches were upstreamed. Could you please double-check my work?

File: srcpkgs/freecad/template [hunk 1]
```diff
@@ -36,17 +36,14 @@ maintainer="yopito <pierre.bourgin@free.fr>"
...
 
 if [ "$XBPS_TARGET_LIBC" = musl ]; then
 	makedepends+=" libexecinfo-devel"
 	LDFLAGS="-lexecinfo"
 fi
 
 post_extract() {
-	# Report exact minor version
-	vsed -i -e "s,^\(set(PACKAGE_VERSION_MINOR\) .*,\1 \"${version#*.}\")," CMakeLists.txt
-
```
Comment (yopito) [inline srcpkgs/freecad/template:49]: don't remove this vsed: will be usefull on next minor minor release

**Review by yopito — CHANGES_REQUESTED** [review]
other points in freecad/template:
* add `python3-pivy` in `makedepends`:
    ```diff
    - qt5-xmlpatterns-devel coin3-doc glew-devel"
    + qt5-xmlpatterns-devel coin3-doc glew-devel python3-pivy"
    ```
* apply this change too:
    ```diff
    -# FreeCAD help: qt5/assistant with SQLite storage
    -depends="python3-matplotlib python3-pyside2 qt5-plugin-sqlite python3-pivy"
    +depends="python3-matplotlib python3-pyside2-webengine python3-pivy"
    ```

**Comment by yopito** [comment]
regarding FreeCAD 0.20 (only): I have some WIP on it, here some points about it:
* help is not generated anymore from source by available (only) from online: see my review of your freecad/template
* `freecad/patches/010-salomesmesh-execinfo.patch` is uneeded anymore
* I've also update libspnav to 1.0 on my side, so additional patch for freecad is needed: could you add these into your commits ?
    * libspnav : update to 1.0: https://gist.github.com/yopito/2757c1a501e2772143814b4a3553cbea
    * freecad: : fix PATH_MAX on musl libc : https://gist.github.com/yopito/fd25b30edcbc762babd9069641a9b86e
* another fix:
    * freecad: : fix build (stderr resource is readonly): https://gist.github.com/yopito/3ac7d4c8fee646171ca900ff17452567
* I have some SIGSEGV on running FreeCAD (x86_64 musl)  when it's using Coin/pivy and shiboken2/pyside2 in 5.15.5 (in WIP) : don't know yet if related to shiboken2/pyside2 in 5.15.5 : see https://github.com/coin3d/pivy/issues/98
   => could you run at least `FreeCAD  -t MeshTestsApp` to see if you have the same trouble ?

**Comment by yopito** [comment]
hello, any news regarding FreeCAD packaging ?

**Comment by luzpaz** [comment]
JFYI, OCCT release v7.6.3 has been [released](https://github.com/Open-Cascade-SAS/OCCT/releases/tag/V7_6_3)

**Comment by yopito** [comment]
hello, since this PR looks stale, I will shortly open another PR focused on freecad 0.20.1 and related updated stuff  (pyside2, libspnav).
Then this one could focus on OCCT solely ?

**Comment by karl-nilsson** [comment]
> hello, since this PR looks stale, I will shortly open another PR focused on freecad 0.20.1 and related updated stuff (pyside2, libspnav). Then this one could focus on OCCT solely ?
> 
> Edit: this is just a proposal, don't hesitate to react.

Please do!
I've been swamped with work lately, I should have some time in the coming weeks to finish this PR.

**Comment by yopito** [comment]
> > hello, since this PR looks stale, I will shortly open another PR focused on freecad 0.20.1 and related updated stuff (pyside2, libspnav). Then this one could focus on OCCT solely ?
> > Edit: this is just a proposal, don't hesitate to react.
> 
> Please do! I've been swamped with work lately, I should have some time in the coming weeks to finish this PR.

OK, thanks. Here it is: #38856 
I suggest to close this PR ?

**Comment by enderger** [comment]
For the musl build, `libexecinfo-devel` should allow it to work. Got about 75% through the build after adding it to the dependency list.

**Comment by Piraty** [comment]
ping

**Comment by luzpaz** [comment]
any more traction here ?

**Comment by luzpaz** [comment]
bump

**Comment by Piraty** [comment]
I assume no. anyone interested in picking up is welcome to open a new PR

**Pull request closed without merging**